internal/runner/jest.go

```diff
@@ -0,0 +1,96 @@
+package runner
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"slices"
+
+	"github.com/buildkite/test-splitter/internal/plan"
+	"github.com/kballard/go-shellquote"
+)
+
+type Jest struct {
+	TestCommand string
+}
+
+func NewJest(testCommand string) Jest {
+	if testCommand == "" {
+		testCommand = "yarn test -- {{testExamples}}"
+	}
+
+	return Jest{
+		TestCommand: testCommand,
+	}
+}
+
+// GetFiles returns an array of file names using the discovery pattern.
+func (r Jest) GetFiles() ([]string, error) {
+	pattern := r.discoveryPattern()
+
+	files, err := discoverTestFiles(pattern)
+	fmt.Printf("%#v\n", files);
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("no files found with pattern %q and exclude pattern %q", pattern.IncludePattern, pattern.ExcludePattern)
+	}
+
+	return files, nil
+}
+
+func (Jest) discoveryPattern() DiscoveryPattern {
+	includePattern := os.Getenv("BUILDKITE_SPLITTER_TEST_FILE_PATTERN")
+
+	if includePattern == "" {
+		includePattern = "**/*.spec.tsx"
```

Comment:
this is the default file pattern `[ "**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)" ]`
https://jestjs.io/docs/configuration#testmatch-arraystring

Reply:
Tried to convert that default into glob syntax that could be supported by zzglob
`**/{__tests__/**/*,*.spec,*.test}.{ts,js,jsx,tsx}`
[Glob test](https://globster.xyz/?q=**%2F%7B__tests__%2F**%2F*%2C*.spec%2C*.test%7D.%7Bts%2Cjs%2Cjsx%2Ctsx%7D&f=%2Fmyapp%2Fsomething%2F__tests__%2Fsome%2Ftest.jsx%2C%2Fmyapp%2Fsomething%2F__tests__%2Fsome%2Fother%2Ftest.js%2C%2Fmyapp%2Fsomething%2F__tests__%2Ftest.tsx%2C%2Fmyapp%2F__tests__%2Fsome%2Fsome%2Ftest.ts%2C%2Fmyapp%2Ftest.spec.ts%2C%2Fmyapp%2Fsomething%2Ftest.spec.tsx%2C%2Fmyapp%2Fsomething%2Fsome%2Ftest.spec.jsx%2C%2Fmyapp%2Fsomething%2Fsome%2Fother%2Ftest.spec.js%2C%2Fmyapp%2Fsomething%2Fsome%2Fnot_a_test.js%2C%2Fmyapp%2Fsomething%2Fsome%2Fmy.test.js)